rest/oidc_api_test.go

```diff
@@ -1,139 +1,865 @@
 package rest
 
 import (
-	"io"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
 	"net/url"
+	"regexp"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/couchbase/sync_gateway/auth"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/gorilla/mux"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"gopkg.in/square/go-jose.v2/jwt"
 )
 
+// The mockAuthServer represents a mock OAuth2 server for verifying OpenID Connect client code.
+// It is not intended to be used as an actual OAuth 2 server. It lacks many features that would
+// be required in a classic implementation. See https://tools.ietf.org/html/rfc6749 to know more
+// about the OAuth 2.0 Authorization Framework specification.
+type mockAuthServer struct {
```

Comment:
Can the mockAuthServer tests be run against the previous OIDC implementation?  It would useful to ensure we aren't changing any behaviour that didn't previously have test coverage.